cloud/scope/clients.go

```diff
@@ -48,14 +47,53 @@ func (c *AzureClients) setCredentials() error {
 	return nil
 }
 
-func getSubscriptionID() (string, error) {
-	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+func getSubscriptionID(subscriptionID string) (string, error) {
 	if subscriptionID == "" {
-		return "", errors.New("error creating azure services. Environment variable AZURE_SUBSCRIPTION_ID is not set")
+		subscriptionID = os.Getenv("AZURE_SUBSCRIPTION_ID")
+		if subscriptionID == "" {
+			return "", errors.New("error creating azure services. Environment variable AZURE_SUBSCRIPTION_ID is not set")
+		}
 	}
 	return subscriptionID, nil
 }
 
-func getAuthorizer() (autorest.Authorizer, error) {
-	return auth.NewAuthorizerFromEnvironment()
+func getAuthorizer(subscriptionID string) (autorest.Authorizer, error) {
+	settings, err := getSettingsFromEnvironment(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	return settings.GetAuthorizer()
+}
+
+func getSettingsFromEnvironment(subscriptionID string) (s auth.EnvironmentSettings, err error) {
```

Comment:
I just realized this is duplicating code from the Azure go-autorest https://github.com/Azure/go-autorest/blob/master/autorest/azure/auth/auth.go#L94, is there any way we can achieve what we need without copying the whole function here? We should avoid maintaining duplicated code since the code might change in the future to add new options and we wouldn't get those updates. cc @devigned 

Sorry I missed this in the first review round.

Reply:
Perhaps, just call the SDK `GetSettingsFromEnvironment`, which will return an `auth.EnvironmentSettings`, and then replace the subscriptionID value.

Reply:
> then replace the subscriptionID value.

Make sure this is only done if and only if `AzureClusterSpec.SubscriptionID != "'` as we still want to allow using the environment variable for subscription ID since the spec field is optional for now.

Reply:
Good call! I knew that was kind of ugly but didn't take the time to investigate the options...